PhotoEnhancer/PhotoEnhancer/Filters/Transform/PerspectiveTransformer.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace PhotoEnhancer.Filters.Transform
+{
+    public class PerspectiveTransformer : ITransformer<PerspectiveParameters>
+    {
+        public Size ResultSize { get; set; }
+        public double N { get; set; }
+        public void Initialize(Size size, PerspectiveParameters parameters)
+        {
+            ResultSize = size;
+            N = parameters.NCoefficient;
+        }
+        public Point? MapPoint(Point p)
+        {
+            var yProgress = (double)p.Y / ResultSize.Height;
+            var cutRatio = ((double)1.0 - N) * ((double)1.0 - yProgress); 
```

Comment:
Если параметр `N` будет в процентах, то эту строку нужно изменить.
Заметим, что переменные `N` и `yProgress` типа `double`, так что преобразования типов лишние.
```suggestion
            var cutRatio = (100 - N) * (1 - yProgress) / 100; 
```